components/Common/Select/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { ChevronDownIcon } from '@heroicons/react/24/outline';
+import * as Primitive from '@radix-ui/react-select';
+import { useId } from 'react';
+import type { FC } from 'react';
+
+import styles from './index.module.css';
+
+type SelectProps = {
+  values: ({ label: string; value: string } | string)[];
+  defaultValue?: string;
+  placeholder?: string;
+  dropdownLabel?: string;
+  label?: string;
+  onChange?: (value: string) => void;
+};
+
+const Select: FC<SelectProps> = ({
+  values = [],
+  defaultValue,
+  placeholder,
+  label,
+  dropdownLabel,
+  onChange,
+}) => {
+  const id = useId();
+
+  return (
+    <div className={styles.select}>
+      {label && (
+        <label className={styles.label} htmlFor={id}>
+          {label}
+        </label>
+      )}
+      <Primitive.Root defaultValue={defaultValue} onValueChange={onChange}>
+        <Primitive.Trigger
+          className={styles.trigger}
+          aria-label={label}
+          id={id}
+        >
+          <Primitive.Value placeholder={placeholder} />
+          <ChevronDownIcon className={styles.icon} />
+        </Primitive.Trigger>
+        <Primitive.Portal>
+          <Primitive.Content className={styles.dropdown}>
+            <Primitive.Viewport>
+              <Primitive.Group>
+                {dropdownLabel && (
+                  <Primitive.Label className={`${styles.item} ${styles.label}`}>
+                    {dropdownLabel}
+                  </Primitive.Label>
+                )}
+                {values.map(item => {
+                  if (typeof item === 'string') {
```

Comment:
I'm not sure about this logic. You could within the map simply have the following

```ts
const value = typeof item === 'string' ? item.value : item;
const label = typeof item === 'string' ? item.label : item;
```

This reduces the amount of duplicated code imo

Reply:
Thanks for the review! I am trying to avoid short ifs but this looks readable and looks way better